File: demo-project/groovy/src/test/java/com/github/gmazzo/buildconfig/demos/groovy/BuildConfigTest.java
package com.github.gmazzo.buildconfig.demos.groovy;

import java.lang.reflect.Modifier;

public class BuildConfigTest extends BuildConfigBaseTest {

    @Override
    protected Object[][] extraCases() {
        return new Object[][] {
            // test properties
            {"aTestValue", TestBuildConfig.TEST_CONSTANT},

            // resources properties
            {"aConstant", BuildResources.A_CONSTANT},
            {"file1.json", BuildResources.FILE1_JSON.getPath()},
            {"file2.json", BuildResources.FILE2_JSON.getPath()},
            {"config/local.properties", BuildResources.CONFIG_LOCAL_PROPERTIES.getPath()},
            {"config/prod.properties", BuildResources.CONFIG_PROD_PROPERTIES.getPath()},

            // custom properties
            {"aValue", Custom.CUSTOM_VALUE},
            {false, Modifier.isPublic(Custom.class.getModifiers())},
        };
    }

}

File: demo-project/kts/src/test/kotlin/com/github/gmazzo/buildconfig/demos/kts/BuildConfigTest.kt
package com.github.gmazzo.buildconfig.demos.kts

import com.github.gmazzo.buildconfig.demos.groovy.BuildConfigBaseTest
import java.util.*
import org.junit.Assert.assertEquals
import org.junit.Test

class BuildConfigTest : BuildConfigBaseTest() {

    private val props by lazy {
        javaClass.getResourceAsStream("/PropertiesBuildConfig.xml").use {
            Properties().apply { loadFromXML(it) }
        }
    }

    override fun extraCases() = arrayOf(
        // test properties
        arrayOf("aTestValue", TestBuildConfig.TEST_CONSTANT),

        // version properties
        arrayOf("1.0.1", myDependencyVersion),

        // resource properties
        arrayOf("aConstant", BuildResources.A_CONSTANT),
        arrayOf("file1.json", BuildResources.FILE1_JSON.path),
        arrayOf("file2.json", BuildResources.FILE2_JSON.path),
        arrayOf("config/local.properties", BuildResources.CONFIG_LOCAL_PROPERTIES.path),
        arrayOf("config/prod.properties", BuildResources.CONFIG_PROD_PROPERTIES.path),

        // custom XML properties
        arrayOf("AAA", props["value1"]),
        arrayOf("BBB", props["value2"]),
        arrayOf("CCC", props["value3"]),
    )

}
